apps/editing-toolkit/editing-toolkit-plugin/coming-soon/coming-soon.php

```diff
@@ -83,15 +96,46 @@ function add_public_coming_soon_to_settings_endpoint_post( $input, $unfiltered_i
  * @return array The updated $templates array
  */
 function add_public_coming_soon_page_template( $templates ) : array {
-	$templates[] = json_decode(
-		// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
-		file_get_contents( __DIR__ . '/coming-soon-page-template.json' ),
-		false
-	);
+	$templates[] = get_coming_soon_page_template();
 	return $templates;
 }
 add_filter( 'vertical_templates', __NAMESPACE__ . '\add_public_coming_soon_page_template' );
 
+// phpcs:disable Generic.CodeAnalysis.UnusedFunctionParameter.FoundBeforeLastUsed
+/**
+ * Adds an editable coming soon page to a site if no coming soon page exists and
+ * a paid plan is purchased
+ *
+ * @param int $blog_id current blog id.
+ * @param int $user_id current user id.
+ * @param int $product_id the id of the plan that was just purchased.
+ */
+function add_coming_soon_page_to_paid_site( $blog_id, $user_id, $product_id ) {
+	if ( ! \WPCOM_Store::is_wpcom_plan( $product_id ) ) {
+		return;
+	}
+
+	// if there is an existing coming soon page don't add a new one.
+	$existing_page_id = (int) get_option( 'wpcom_public_coming_soon_page_id', 0 );
+	if ( $existing_page_id ) {
+		return;
+	}
+
+	$coming_soon_template = get_coming_soon_page_template();
+	$new_page_id          = wp_insert_post(
+		array(
+			'post_title'     => $coming_soon_template->post_title,
+			'post_content'   => $coming_soon_template->post_content,
+			'post_status'    => 'publish',
+			'post_type'      => 'page',
+			'comment_status' => 'closed',
+			'ping_status'    => 'closed',
+		)
+	);
+	add_option( 'wpcom_public_coming_soon_page_id', $new_page_id );
+}
+add_action( 'subscription_changed', __NAMESPACE__ . '\add_coming_soon_page_to_new_site', 10, 3 );
```

Comment:
Not sure yet if we'd only want to set this page up for paid sites.

My opinion is that we create it for all sites, so we can open up upgrade opportunities.

I think it's okay for now since it's a one line change to add it to all (is that right?)